### PR TITLE
JENKINS-66040: fix in-progress notifications for non-standard project recognizers

### PR DESCRIPTION
**Pull request by dee-kryvenko**
See https://issues.jenkins.io/browse/JENKINS-66040

### Your checklist for this pull request

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

**Review by bitwiseman — CHANGES_REQUESTED** [review]
Please add a test that verifies this change.

**Comment by dee-kryvenko** [comment]
@bitwiseman that wasn't easy for a beginner to figure out given no tests previously existed for this piece of functionality - but I think I pulled it off. Hopefully that will do.

**Review by bitwiseman — APPROVED** [review]

**Comment by bitwiseman** [comment]
@dee-kryvenko 
Great job figuring out tests!

**Comment by dee-kryvenko** [comment]
Thanks @bitwiseman !

File: src/test/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotificationsTest.java [hunk 1]
```diff
@@ -50,7 +94,114 @@ public void noInappropriateFirstCheckoutCompletedInvisibleAction() throws Except
...
+
+        owner.scheduleBuild2(0).getFuture().get();
+        owner.getComputation().writeWholeLogTo(System.out);
+        assertThat(owner.getIndexing().getResult(), is(Result.SUCCESS));
+        r.waitUntilNoActivity();
+        WorkflowJob master = owner.getItem("master");
+        WorkflowRun run = master.getLastBuild();
+        run.writeWholeLogTo(System.out);
+        assertThat(run.getResult(), is(Result.SUCCESS));
+        assertThat((List<Action>) run.getAllActions(), not(hasItem(instanceOf(FirstCheckoutCompletedInvisibleAction.class))));
```
Comment (jglick) [inline src/test/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotificationsTest.java:182]: This is failing with https://github.com/jenkinsci/workflow-multibranch-plugin/pull/582 but I cannot figure out what the purpose of this assertion (and test case) is. https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/130 as written seems to make sense, and so does the main source code change here, but what is the test supposedly proving?

**Pull request merged**